pkg/apis/bundle/v1alpha1/patch_template_types.go

```diff
@@ -19,6 +19,35 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// ObjectSelector is used for identifying Objects on which to apply the patch template.
+type ObjectSelector struct {
+	// Kinds represent the Kinds to match.
+	Kinds []string `json:"kinds,omitempty"`
+
+	// Names represent the metadata.names to match.
+	Names []string `json:"names,omitempty"`
+
+	// Annotations contain key/value pairs to match. An empty string value matches
+	// all annotation-values for a particular key.
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Labels contain key/value pairs to match. An empty string value matches
+	// all label-values for a particular key.
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Namespaces to match.
+	Namespaces []string `json:"namespaces,omitempty"`
+
+	// NamespacedOnly matches only if the type of object is namespaced (i.e.,
+	// is not a cluster-wide resource).
+	NamespacedOnly *bool `json:"namespacedOnly,omitempty"`
+
+	// NegativeMatch invert the match. By default, the ObjectSelector will include
+	// objects matching all of the criteria above. This flag indicates that objects
+	// NOT matching the criteria should be included instead.
+	NegativeMatch *bool `json:"negativeMatch,omitempty"`
```

Comment:
I don't think this needs to be a bool-pointer. Since not specifying this is precisely equivalent to false